siriuspy/siriuspy/devices/bpm_eq.py

```diff
@@ -0,0 +1,581 @@
+        self.set_gains(self.data['gains_init'])
+        self._log('Acquisition Finished!')
+
+    def _do_acquire(self):
+        if self._acq_strategy == self.AcqStrategies.AssumeOrder:
+            self.set_gains(self.MAX_MULTIPLIER)
+        else:
+            gains = _np.full((len(self.bpms), 4, 2), self.MAX_MULTIPLIER)
+            gains[:, :, 1] *= self._acq_inverse_reduced_gain
+            self.set_gains(gains)
+        self.data['acq_strategy'] = self._acq_strategy
+        self.data['acq_inverse_reduced_gain'] = self._acq_inverse_reduced_gain
+
+        # acquire antennas data in FOFB rate
+        self._log('Preparing BPMs')
+        ret = self.cmd_mturn_acq_abort()
+        if ret > 0:
+            self._log(f'ERR: BPM {ret-1} did not abort previous acquistion.')
+            return
+
+        self.mturn_reset_flags_and_update_initial_timestamps()
+        ret = self.mturn_config_acquisition(
+            nr_points_after=self._acq_nrpoints, nr_points_before=0,
+            acq_rate='FOFB', repeat=False, external=True)
+        if ret > 0:
+            self._log(f'ERR: BPM {ret-1} did not start acquistion.')
+            return
+
+        self._log('Waiting BPMs to update')
+        ret = self.mturn_wait_update(timeout=self._acq_timeout)
+        if ret > 0:
+            self._log(f'ERR: BPM {ret-1} did not update in time.')
+            return
+        elif ret < 0:
+            self._log(f'ERR: Problem with acquisition. Error code {ret}')
+            return
+        self._log('BPMs updated.')
+
+        self._log('Acquiring data.')
+        fsamp = self.get_switching_frequency(1)
+        fswtc = self.get_sampling_frequency(1)
+        self.data['freq_switching'] = fsamp
+        self.data['freq_sampling'] = fswtc
+        if None in {fsamp, fswtc}:
+            self._log('ERR: Not all BPMs are configured equally.')
+            return
+        elif fsamp % fswtc:
+            self._log('ERR: Sampling freq is not multiple of switching freq.')
+            return
+
+        posx_gain, posy_gain = [], []
+        for bpm in self.bpms:
+            posx_gain.append(bpm.posx_gain)
+            posy_gain.append(bpm.posy_gain)
+        self.data['posx_gain'] = _np.array(posx_gain)
+        self.data['posy_gain'] = _np.array(posy_gain)
+        self.data['gains_acq'] = self.get_current_gains()
+
+        _time.sleep(0.1)
+        self.data['antennas'] = _np.array(
+            self.get_mturn_signals()).swapaxes(1, 0)
+
+    # --------- Methods for processing ------------
+
+    def process_data(self):
+        """Process data."""
+        if 'antennas' not in self.data:
+            self._log('ERR:There is no data to process. Acquire first.')
+            return
+        mean_antennas = self.calc_switching_levels(**self.data)
+        self.calc_gains(mean_antennas)
+        self.estimate_orbit_variation()
+
+    def calc_switching_levels(
+            self, antennas, fsamp=4, fswtc=2, acq_strategy=0,
```

Comment:
Shouldn't the defaults here be `fsamp=4` and `fswtc=1`, to more accurately match the machine setup?

Reply:
no, because the switching frequency returned by the BPMs is 2x the real switching frequency, and that's the value that I always assume in my code.

Reply:
https://github.com/lnls-sirius/dev-packages/blob/d66a4bc16273482f757cff496acf61a3ce570a5b/siriuspy/siriuspy/devices/bpm.py#L212

Isn't that handled there already?

Reply:
@fernandohds564

Reply:
right, thanks!

Reply:
done!